transport/http/inbound.go

```diff
@@ -154,3 +155,16 @@ func (i *Inbound) Addr() net.Addr {
 
 	return listener.Addr()
 }
+
+// Introspect returns the state of the inbound for introspection purposes.
+func (i *Inbound) Introspect() introspection.InboundStatus {
```

Comment:
Doesn't this make it impossible to implement a transport outside the library, since the interface is in `internal`?